### ISSUE TITLE
to add ?

### ISSUE THREAD

**Issue by Jordan-TE** [favll/pogom#199]
i found this on reddit : 
https://www.reddit.com/r/pokemongodev/comments/4wxteh/i_implemented_tbterras_spawntracker_into/
can lower the scan time and the account needed. 
sadly i don't have time to make a pr, some people made it work with pokemongo MAP so it should be possible here i guess

**Comment by lisandro52** [comment]
I tried implementing that yesterday, adding all my known spawnPoints to the `LOCATIONS` array on the `config.json`, but pogom's Scanner isn't prepared for that (it was way too slow compared to single location method). What I'm thinking is having a new table (filled with the known spawnPoints and their lat/lng), so each time a new step is going to be scanned, you first check if it's included in that table. If it's there, you send the request; if it isn't you go on to the next step.

Next thing to talk about would be how and when to fill that new table, supposing that spawnPoints can change over time (not every day, as far as we know). One solution would be executing a job once every 24 hours that scans all locations without checking them for 1.5 hours or so, to be sure we are getting *all* the (new?) spawnPoints. Once that's done, update the `spawnPoints table` with the new info and go back to the check-me-first method.

**Comment by lisandro52** [comment]
By the way, to get all the **different** spawnPoints you have on your database, the query would be
```
SELECT spawnpoint_id, latitude, longitude
FROM pokemon
GROUP BY spawnpoint_id
```

**Comment by compuguy** [comment]
@Jordan-TE I think we would need to get the original spawntracker code, because what was posted on reddit is an implementation for PokemonGo-Map. The original spawntracker code is here: https://github.com/TBTerra/spawnTracker

**Comment by Jones1337** [comment]
We urgently need this :+1:

**Comment by khag7** [comment]
This is the only thing missing from your app that is keeping me on PokemonGo-Map. My efficiency has tripled using spawn point based scanning.

Ideally your web interface should have a "upload spawn points" feature. I can dump spawnpoint_id and  disappear time from an existing database as a CSV. Your app would take the uploaded file and use it to seed the initial searching. Spawn point id is an s2 level 20 cell token that can be converted to lat and lon via s2sphere library. Store the spawn point id and the spawn time (minute of the disappear time -15) in a db table called spawnpoints and start scanning based on those locations. All new spawnpoints found or imported/uploaded later will be added to the db in the same way.

Make these changes and you'll have the easiest to use, fastest scanning map app available. Throw in webhooks & the code from pokealert and you can check notifications off your list too. This is crucial.

**Comment by nborrmann** [comment]
The main problem with this is, detecting whether or not a region has been fully scanned for an hour and if we can swap the search algo.

**Comment by khag7** [comment]
The new scanning method can be implemented and let the user choose which to do. Then in the future we can implement a smart auto-swirhching as a 3rd option to let the app choose to switch between the two options as needed. Maybe once every other day it can rescan an area for new spawnpoints.

Honestly I'd never rescan for points, id leave it on spawnpoints based scanning always. Then id run a spawnpoints scanner script separately and periodically import spawnpoints to the pogom db. 

People in the same geographical area will be able to share spawn lists with each other. Basically saying "hey I scanned a new region of our town, here's the list for it". Not every user needs to do a  "slow" scan. Spawnpoints are the same for everyone. Once an area is scanned by one person, others can import the lisr. Saves ton of time and more importantly it greatly reduces requests to niantic api.

**Comment by kamieniarz** [comment]
Can't you just create new script for collecting spawnpoints for 1-2h of running? After that we can run main script which will base on spawnpoints collected from first script. I hope I'm clear enough :D

**Comment by nborrmann** [comment]
I won't push any half-baked solution into master, and I don't have too much time right now, so this may take a while.

Scanning an area and making sure you catch all 30m, 45m, 60m and double spawns is anything but trivial. Whatever you're using now probably doesn't recognize any spawns >15m.

**Comment by khag7** [comment]
Right. That's the hard part, getting an accurate list of spawnpoint_id's & their associated spawn times.

But a lot of people already have this data! Your scanner doesn't need to pull this info. You just need to give users the option of providing that data from their existing weeks of scanning.

I have over 100k spawnpoint_id's and I know exactly what time those spawns all expire. I want to provide a file to pogom with that information. Pogom will scan each of those points 14.5 minutes before despawn time. 97% of those spawned pokemon will have only been there for 30 seconds. Yes, its possible that there are 3% of those points which actually had spawned more than 30 seconds ago, but I don't really need to know that. I am not scanning to go run out and collect these pokemon, I am storing a database of which pokemon spawn at what points so I can do heatmapping of my entire region. Don't get so hung up on timing, its not as important as you think.

**Comment by nborrmann** [comment]
I'm kind of a perfectionist though.

**Comment by Faskill** [comment]
Any advance on this?
